src/components/icon/icon.ts

```diff
@@ -0,0 +1,178 @@
+import {
+  AfterContentChecked,
+  ChangeDetectorRef,
+  Component,
+  Directive,
+  ElementRef,
+  HostBinding,
+  Input,
+  OnChanges,
+  OnInit,
+  Renderer,
+  SimpleChange,
+  ViewEncapsulation,
+} from 'angular2/core';
+
+import {NgClass} from 'angular2/common';
+
+import {
+  MdIconRegistry,
+} from './icon-registry';
+
+@Component({
+  template: '<ng-content></ng-content>',
+  selector: 'md-icon',
+  styleUrls: ['./components/icon/icon.css'],
+  host: {
+    'role': 'img',
+  },
+  directives: [NgClass],
+  encapsulation: ViewEncapsulation.None,
```

Comment:
You should be able to add 
```ts
changeDetection: ChangeDetectionStrategy.OnPush,
```
without changing any behavior (for perf boost for change detection)